packages/edit-site/src/store/private-actions.js

```diff
@@ -13,39 +13,56 @@ import { store as editorStore } from '@wordpress/editor';
 export const setCanvasMode =
 	( mode ) =>
 	( { registry, dispatch } ) => {
-		const isMediumOrBigger =
-			window.matchMedia( '(min-width: 782px)' ).matches;
-		registry.dispatch( blockEditorStore ).__unstableSetEditorMode( 'edit' );
-		const isPublishSidebarOpened = registry
-			.select( editorStore )
-			.isPublishSidebarOpened();
-		dispatch( {
-			type: 'SET_CANVAS_MODE',
-			mode,
-		} );
-		const isEditMode = mode === 'edit';
-		if ( isPublishSidebarOpened && ! isEditMode ) {
-			registry.dispatch( editorStore ).closePublishSidebar();
-		}
-
-		// Check if the block list view should be open by default.
-		// If `distractionFree` mode is enabled, the block list view should not be open.
-		// This behavior is disabled for small viewports.
-		if (
-			isMediumOrBigger &&
-			isEditMode &&
+		const switchCanvasMode = () => {
+			const isMediumOrBigger =
+				window.matchMedia( '(min-width: 782px)' ).matches;
+			registry.dispatch( blockEditorStore ).clearSelectedBlock();
```

Comment:
When top toolbar is ON and we exit edit mode to view mode there is a short moment when the header is removing the top toolbar and _then_ the animation moves forward. Can we do it in an unblocking way? Should it be already unblocking? 🤷🏻

Reply:
I'm not sure that's entirely what you meant but I noticed that we're triggering a big number of actions when switching canvas mode, so I batched everything in 6c58262cf3

I think the animation is smoother with that.

Reply:
Yes this is exactly what I meant :) The animation doesn't look like it's waiting on something to be over before starting, it just goes now. 👏🏻